src/util.h

```diff
@@ -115,6 +115,11 @@ bool Truncate(const string& path, size_t size, string* err);
 #endif
 
 #ifdef _WIN32
+/// UTF-8 converters
+std::wstring Utf8ToWide(const std::string& s);
+std::string WideToUtf8(const std::wstring& s);
```

Comment:
WStringPiece

Reply:
Would returning StringPiece/WStringPiece improve the situation? If so, one can decide at a later instance if the value must be used as a string or char*, without always starting as a string and then copying it's content out, but still get the benefits of not dealing with char pointers.

Reply:
StringPiece is non-owning IIRC so you shouldn't return it.

Reply:
You need to have two versions of these functions.

One that returns the string type. And another that allows the end user to pass in a pointer to the buffer to  buffer into which the converted data should go.

You can avoid the second one by modifying the code where it would be used to call the windows api directly though, which would be much better.

Reply:
OK, thanks for the clarification!